generator/import.go

```diff
@@ -1,49 +1,54 @@
 package generator
 
 import (
-	"log"
-	"sort"
+	"go/types"
 	"strings"
 )
 
+// Imports indexes imports by package path and alias so that all imports have a
+// unique alias, and no package is included twice.
+type Imports struct {
+	ByAlias   map[string]Import
+	ByPkgPath map[string]Import
+}
+
+func newImports() Imports {
+	return Imports{
+		ByAlias:   make(map[string]Import),
+		ByPkgPath: make(map[string]Import),
+	}
+}
+
 // Import is a package import with the associated alias for that package.
 type Import struct {
-	Alias string
-	Path  string
+	Alias   string
+	PkgPath string
 }
 
 // AddImport creates an import with the given alias and path, and adds it to
 // Fake.Imports.
-func (f *Fake) AddImport(alias string, path string) Import {
+func (i *Imports) Add(alias string, path string) Import {
+	// TODO: why is there extra whitespace on these args?
```

Comment:
Not sure I understand; which white space / args are you referring to?

Reply:
`path` and `alias` are both passed to `strings.TrimSpace` so I assume there is either leading or trailing whitespace on those strings, but it is not clear to me where that would come from. It seems like this is called with values from inspecting `go/types`, which I would assume do not contain extra whitespace.

Reply:
I guess I am just always defensive with function arguments (particularly when exported).